lib/core/network/models/transaction_model.dart

```diff
@@ -1,72 +1,336 @@
 import 'package:equatable/equatable.dart';
 
-class TransactionModel extends Equatable {
+sealed class TransactionModel extends Equatable {
   final DateTime timestamp;
-  final String? transactionId;
   final String account;
   final String actionName;
   final int blockNumber;
+  final String actor;
+  final String? transactionId;
   final Map<String, dynamic> data;
 
   const TransactionModel({
     required this.data,
     required this.actionName,
     required this.account,
     required this.timestamp,
-    required this.transactionId,
     required this.blockNumber,
+    required this.actor,
+    required this.transactionId,
   });
 
   @override
-  List<Object?> get props => [transactionId];
+  List<Object?> get props => [actionName, data, account, timestamp, blockNumber];
 
   factory TransactionModel.fromJson(Map<String, dynamic> json) {
-    return TransactionModel(
-      account: json['act']['account'],
-      actionName: json['act']['name'],
-      data: json['act']['data'],
-      timestamp: parseTimestamp(json['@timestamp']),
-      transactionId: json['trx_id'],
-      blockNumber: json['block_num'],
-    );
-  }
+    final act = json['act'];
+    final actionName = act['name'];
+    final account = act['account'];
+    final data = act['data'];
+    final timestamp = parseTimestamp(json['@timestamp']);
+    final blockNumber = json['block_num'];
+    final actor = act['authorization'].first['actor'];
+    final transactionId = json['trx_id'];
 
-// factory TransactionModel.fromJsonMongo(Map<String, dynamic> json) {
-//   return TransactionModel(
-//     account: json['act']['account'],
-//     actionName: json['act']['name'],
-//     data: json['act']['data'],
-//     timestamp: parseTimestamp(json['@timestamp']),
-//     transactionId: json['trx_id'],
-//     blockNumber: json['  block_num'],
-//   );
-// }
-
-// static TransactionModel? fromTransaction(GenericTransactionModel genericModel) {
-//   if (genericModel.transaction.isTransfer) {
-//     final action = genericModel.transaction.actions.first;
-//     final data = action.data;
-//     final String? from = data?['from'];
-//     final String? to = data?['to'];
-//     final String? quantity = data?['quantity'];
-//     final String memo = data?['memo'] ?? "";
-//     if (from != null && to != null && quantity != null) {
-//       return TransactionModel(
-//         from: from,
-//         to: to,
-//         quantity: quantity,
-//         memo: memo,
-//         timestamp: genericModel.timestamp ?? DateTime.now().toUtc(),
-//         transactionId: genericModel.transactionId,
-//       );
-//     }
-//   }
-//
-//   return null;
-// }
+    switch (actionName) {
```

Comment:
action detection is account + actioname, in general. 

for transfer, it's likely going to be a token transfer, but we need to check all data params to ensure that that's the case, since we don't know all existing token contracts. So we need to take a bit of a guess there.

I would recommend outsourcing the detection of the type to some logic outside. 

example: 

you could have transfer on token.seeds, or on hypha.hypha, or on eosio.token (TLOS/EOS), ..

However, there's nothing stopping me from defining my contract to have a transfer action, with different parameters. 

So I could have 
mycontract -> transfer(whatever)

And this would not be a normal token transfer action. 

I think for transfer, we can take all params, and if they're all as they need to be for a token transfer, we say it's a transfer action. 

Other actions only exist on hypha DAO, so we need to define the contract somewhere, as a known DAO contract. 

We need to then compare accountname to list of known dao contracts

initially the list is just ['dao.hypha'] but could expand later. 

Was also wondering about scams but realized that if someone deploys a malicious contract they can do all sorts of damage and it's not really related to us, or within our scope to deal with. 

We could also limit transfer to know trusted contracts eventually...